app/Decsys/Services/WebhookService.cs

```diff
@@ -84,13 +85,19 @@ private static bool CheckIsValid(PageNavigation webhookNavigation, PageNavigatio
     /// Posts webhook data to a given URL.
     /// </summary>
     /// <param name="payload">Payload to Post</param>
-    /// <param name="webhook">Webhook to Post to</param>
+    /// <param name="webhook">An instance of WebhookModel containing the CallbackUrl and a Secret for HMACSHA256 hashing</param>
     private async Task SendWebhook(WebhookModel webhook, PayloadModel payload)
     {
         var json = JsonConvert.SerializeObject(payload);
 
         var content = new StringContent(json, Encoding.UTF8, "application/json");
-        if (webhook.SecretHash is not null) content.Headers.Add("X-Decsys-Signature", webhook.SecretHash);
+        
+        using var hmac = new HMACSHA256(Encoding.UTF8.GetBytes(webhook.Secret));
+        
+        if (webhook.Secret is not null)   content.Headers.Add("X-Decsys-Signature",
+            Encoding.UTF8.GetString(
+                hmac.ComputeHash(
+                    Encoding.UTF8.GetBytes(json))));
```

Comment:
To fix the warnings we should put the hmac initialisation inside the null check condition
```suggestion
         if (webhook.Secret is not null)
         {
            using var hmac = new HMACSHA256(Encoding.UTF8.GetBytes(webhook.Secret));
            content.Headers.Add("X-Decsys-Signature",
                Encoding.UTF8.GetString(
                    hmac.ComputeHash(
                        Encoding.UTF8.GetBytes(json))));
          }
```